From: Miles Chen <miles.chen@mediatek.com>
Date: Mon, 13 Sep 2021 11:06:25 +0800
Subject: scripts/sorttable: riscv: fix undelcred identifier 'EM_RISCV' error
Patch-mainline: Submitted - 2021-09-13 - 20210913030625.4525-1-miles.chen@mediatek.com
References: e8965b25-f15b-c7b4-748c-d207dda9c8e8@i2se.com

Fix the following build break by adding a conditional definition
of EM_RISCV in order to allow cross-compilation on machines
which do not have EM_RISCV definition in their host.

build break log from [1]:
scripts/sorttable.c:352:7: error: use of undeclared identifier 'EM_RISCV'

[1] https://lore.kernel.org/lkml/e8965b25-f15b-c7b4-748c-d207dda9c8e8@i2se.com/

Cc: Jisheng Zhang <jszhang@kernel.org>
Reported-by: Stefan Wahren <stefan.wahren@i2se.com>
Fixes: 54fed35fd393 ("riscv: Enable BUILDTIME_TABLE_SORT")
Signed-off-by: Miles Chen <miles.chen@mediatek.com>
Acked-by: Michal Kubecek <mkubecek@suse.cz>
---
 scripts/sorttable.c | 4 ++++
 1 file changed, 4 insertions(+)

--- a/scripts/sorttable.c
+++ b/scripts/sorttable.c
@@ -54,6 +54,10 @@
 #define EM_ARCV2	195
 #endif
 
+#ifndef EM_RISCV
+#define EM_RISCV	243
+#endif
+
 static uint32_t (*r)(const uint32_t *);
 static uint16_t (*r2)(const uint16_t *);
 static uint64_t (*r8)(const uint64_t *);
